packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -392,11 +391,10 @@ class RenderWrap extends RenderBox with ContainerRenderObjectMixin<RenderBox, Wr
     while (child != null) {
       final double childWidth = child.getMaxIntrinsicWidth(double.infinity);
       final double childHeight = child.getMaxIntrinsicHeight(childWidth);
-      if (runWidth + childWidth > width) {
-        height += runHeight;
-        if (runCount > 0)
-          height += runSpacing;
-        runCount += 1;
+      final double spaceWidth = childCount > 0 ? spacing : 0;
+      // child count check for situation when width of first child in run over available width
+      if (runWidth + childWidth + spaceWidth > width && childCount > 0) {
+        height += runHeight + runSpacing;
```

Comment:
what if there is only one run? In that case wouldn't we want to exclude the runSpacing?

Reply:
If there's only one run `runWidth + childWidth + spaceWidth > width` in line 396 wouldn't be true and this wouldn't execute, so that case should be covered.

Reply:
Would be good to confirm this by adding a test case, though.

Reply:
If there is only one run, we should not use runSpacing because it documentation tell us - space to place between the runs themselves in the cross axis. If we run only ones we have only one run in all the cases.
Code in 'if(width-of-current-run-more-than-available and more-then-first-child)' covered a transfer to new run, but before transfer we counting runHeight and use them in the end of counting intrinsic hight.
height += runHeight;
If we have no transfer - intrinsic hight is hight of the only one run.